### PR TITLE
Issue Fixes

### LINKED ISSUE #23651: Reinforced vending machine breaks into glass instead of armor
The reinforced vending machine only drops glass when broken into; this should be changed to heavy steel armor, concrete debris or reinforced glass debris. It should also have a unique broken form different from the ordinary vending machine.

### LINKED ISSUE #23947: Small storage units gate issue
Game version: Currently 0.C-28092

Operating system: Windows 7

Tiles or curses: Tiles

Mods active: More Survival Tools, No Explosive Zombies, No Fungal Monsters, StatsThroughSkill, Vehicle Additions Pack, Tanks and Other Vehicles, Bionics Systems Mod, Boats, Icecoon's Arsenal, Filthy Clothing, Makeshift Items Mod, More Buildings, No Flaming Weapons, Disable Mutagens, Disable NPC Needs, No Antique Firearms, No Fictional Guns, NPC Traits

A very minor issue:
In default state, 4 gates leading to different compartments are closed. By each one of them there's a winch. 2 of the winches open connected gates as they should, but 2 report that "The gate is closed!" despite it being closed already. Using those winches again, open them correctly.

### Steps to reproduce the behavior
Visit aforementioned storage units, open/close gates.

Pic of location in question:
storage

### LINKED ISSUE #24139: Log and Sod/Thatched Roof yields nails when removed
Game version: 0.C-28639-g36376e4

Operating system: Windows 10

Tiles or curses: Tiles

Mods active: Dark Days Ahead, Arcana and Magic Items, Big River Building Pack, Medieval and Historic Content, Mining Mod, More Survival Tools, No Fungal Monsters, StatsThroughSkills Revised, Vehicle Additions Pack, Tanks and Other Vehicles, Bionic Systems Mod, Folding Parts Pack, Icecoon's Arsenal, Filthy Clothing, Makeshift Items Mod, More Buildings, Mutant NPCs, NPC Traits, Extended Realistic Guns, Boats

### Expected behavior
Destroying Log and Sod/Thatched Roof and causing rubble to fall should only yield wood-related items when cleared.

### Actual behavior
Currently, tearing down Log and Sod/Thatched Roofs and causing rubble to fall will yield Wood Splinters and Nails even though there are no Nails in the recipe.

### Steps to reproduce the behavior
Build several roof tiles and tear one down with a pickaxe or similar tool to cause rubble to collapse. Clear the rubble with a shovel and both splintered wood and Nails will be left when you cleared the rubble.

### LINKED ISSUE #24654: Lab Rock Floor
Should labs have different, more hightec, cool looking floor?
Cause now it using same floor as in caves or mines, and i do not think this is right. 
Sorry if it is not an issue. I just found that to be weird looking, then i changed rock floor tiles in my tileset to look more like a rock in a cave with cracks and dirt, and not black and clean as fresh asphalt.
So now my caves and mines look awesome, but labs look like caves. :\

### LINKED ISSUE #24675: (Bicycle shop's) vehicle placement may damage the building
Game version: O.C-30062

Operating system: Windows 7 

Tiles or curses: Tiles

Mods active: More Survival Tools, StatsThroughSkill, Vehicle Additions Pack, Tanks and Other Vehicles, Bionics Systems Mod, Boats, Icecoon's Arsenal, Filthy Clothing, Makeshift Items Mod, More Buildings, No Flaming Weapons, Disable Mutagens, Disable NPC Needs, No Antique Firearms, No Fictional Guns, NPC Traits

### Expected behavior
Vehicles in locations like bicycle shop are positioned in a reasonable way that wouldn't be out of place IRL.

### Actual behavior
Vehicles have a chance to channel the Kool-Aid Man:

uhoh

### PR DESCRIPTION
**Pull request by cosmiccoincidence**
This fix closes #23651, closes #23947, closes #24139, closes #24654, and closes #24675.

#23651 
- Adds sheet_metal and steel_chunk to reinforced_vending_machine bash list list.

#23947 
- Moves the bottom 2 winches similar to the top 2 and puts columns in their empty spots similar to the north side of the building.

#24139 
- Adds a new terrain (primitive floor) and uses that new floor type as the floor for the log and sod roof.

#24654 
- Changes rock floor to concrete floor in labs.

#24675 
- Moved one bike spawn over to prevent it from spawning in the wall.

### CODE REVIEW

**Comment by cosmiccoincidence** [comment]
Not sure why there is conflicting files (lab_floorplans.json). All I did with those files was replace all "rock_floor"s with "thconc_floor"s. I didn't see anything overlapping in the palettes, nor got any errors.

**Comment by BrianLefler** [comment]
This will also require a change to mapgen.cpp because there are hardcoded references there.

Basically replace every references to t_rock_floor with t_ thconc_floor from around line 2823 (the else if terrain_type == "lab" .... block) down until around line 3643 (the else if terrain_type = "outpost" block).

If you don't make that change, the hardcoded lab segments won't change their floors, and more importantly stairs won't work properly in the new json lab segments.

**Comment by BrianLefler** [comment]
Also for the t_rock_floor references inside of mapgen.cpp's set_science_room() function around line 7416.

And I think that should do it for the cpp changes.  I believe it can just be a straight search & replace within those regions, should not require changing any of the logic.

**Comment by cosmiccoincidence** [comment]
I havnt done really any work with c++, but can look into making those changes. Also Don't know where mapgen.cpp is stored.

**Comment by BrianLefler** [comment]
The path to the file is: src/mapgen.cpp

I believe if the change you make is just a straight replacement of t_rock_floor with t_thconc_floor (no quote marks) anywhere it appears between lines 2823 and 3643 then everything should work and you won't need to know anything about code.  Happy to take a look and confirm.

**Comment by BrianLefler** [comment]
Looks good to me.

**Comment by cosmiccoincidence** [comment]
Not sure why there is a conflicting file still. Only thing different in the conflict is part of my change of floors.

**Comment by Night-Pryanik** [comment]
Please use magic github words like `Closes #xxxxx`, where xxxxx is the number of an issue. Use the whole sentence for every issue meant to be closed. It will auto-close the issues on PR merging.

File: data/json/furniture.json [hunk 1]
```diff
@@ -2190,7 +2190,10 @@
             "sound_fail": "whack!",
             "furn_set": "f_vending_o",
             "items": [
-                { "item": "glass_shard", "count": [1, 3] }
+                { "item": "glass_shard", "count": [1, 3] },
+                { "item": "reinforced_glass_pane", "count": [0, 1] },
```
Comment (Night-Pryanik) [inline data/json/furniture.json]: I'd say if it's `glass breaking!` then there couldn't be any intact reinforced glass pane.

File: data/json/mapgen/lab/lab_floorplans.json [hunk 2]
```diff
@@ -51,7 +51,7 @@
     }
   },
   {
-    "//": "military bunker with 1-in-10 chance of the doors being smashed in by a hulk.",
```
Comment (Night-Pryanik) [inline data/json/mapgen/lab/lab_floorplans.json]: Why did you remove the extended comments? They are comments, after all.

**Comment by Night-Pryanik** [comment]
Jenkins rebuild

File: data/json/mapgen/lab/lab_floorplans.json (hunk 2 above)
Comment (cosmiccoincidence) [reply data/json/mapgen/lab/lab_floorplans.json]: This change was not me, If you check CleverRaven's master you will see the change was made recently (I believe when metal doors were switched out for frosted glass doors). The reason you see the change here is because my lab_floorplans.json was not updated the first commit but I recommited my changes with the new version of this file.

File: data/json/furniture.json (hunk 1 above)
Comment (cosmiccoincidence) [reply data/json/furniture.json]: Makes sense. Fixed.

**Comment by Night-Pryanik** [comment]
By the way, you can add `deconstruct` field to the reinforced vending machines, and add reinforced glass pane to that field.

**Comment by cosmiccoincidence** [comment]
Using Closes #xxxxx, does that go anywhere or does that have to be in the title of the PR?

Also do you think I should add deconstruct option to reinforced vending machines? If I do Id have to with the other vending machines as well..

**Comment by DracoGriffin** [comment]
>Using Closes #xxxxx, does that go anywhere or does that have to be in the title of the PR?

Closes/Fixes/Resolves are all interchangeable and ideally in the first post (which you've already done). Once this is merged, it will automatically close those Issues, so you've done what'd needed (as far as I can tell).

**Comment by Night-Pryanik** [comment]
> Using Closes #xxxxx, does that go anywhere or does that have to be in the title of the PR?

That has to be in your opening post.

> Also do you think I should add deconstruct option to reinforced vending machines? If I do Id have to with the other vending machines as well..

Yes, I think this is plausible, but that could be done in some other PR.

**Comment by cosmiccoincidence** [comment]
Thanks for your patience with me, clearly I'm still learning. =]

**Comment by Night-Pryanik** [comment]
You are welcome.

**Comment by cosmiccoincidence** [comment]
'test'

**Comment by cosmiccoincidence** [comment]
>By the way, you can add deconstruct field to the reinforced vending machines, and add reinforced glass pane to that field.

Around "deconstruct" of your comment. I presume that's some in-comment code thing. Is that necessary for the #xxxxx and how do I do it?

**Comment by cosmiccoincidence** [comment]
Closes #24650


test ignore

**Comment by cosmiccoincidence** [comment]
'test'

**Comment by cosmiccoincidence** [comment]
Ok, that was the last issue fix I'm adding to this PR. I don't want to make this PR too complicated.

**Comment by Faalagorn** [comment]
Last time I asked in Discord whether more smaller PRs are welcome or one bigger and generally got a reply that smaller PRs *should* be preferable which I personally like too, but I really like the small fixes you nailed in this :).
